process/nodemanager/node_manager.go

```diff
@@ -92,12 +95,12 @@ func (nm *NodeManager) InitAppDHT(slotsPerNode int) error {
 // InitialiseNode will be called once the dht is initialised.
 // It will help setting up the connections and initialise the datastores
 func (nm *NodeManager) InitialiseNode() error {
-	slots := nm.dhtMgr.GetSlotsForNode(nm.selfNodeID)
-	if len(slots) <= 0 {
-		return dht.ErrNotInitialised
+	shards := nm.dhtMgr.GetAllShardsForNode(nm.selfNodeID)
+	if len(shards) <= 0 {
+		return dht.ErrDHTNotInitialised
 	}
 
-	if err := nm.dsmgr.InitialiseDataStores(slots); err != nil {
+	if err := nm.dsmgr.InitialiseDataStores(shards); err != nil {
```

Comment:
* This method spins one go-routine to schedule jobs and we are not imposing constrain to keep it only one.
* We are just re-initialising WAL for each shard, we need to do that for new shards only.
* When are we transferring the existing shard data to the newly added instance? That would also add too much load to the system.

Reply:
> This method spins one go-routine to schedule jobs and we are not imposing constrain to keep it only one.

For now since we are calling it only once during the initialisation, it should be fine.

> We are just re-initialising WAL for each shard, we need to do that for new shards only.

Internally, incase the wal log already exists, we start from where we left off. You can see this in wal library init method. 
```go
wal.openExistingOrCreateNew(opts.LogDir)
```
For new shards, it will create new wal from scratch.

> When are we transferring the existing shard data to the newly added instance? That would also add too much load to the system.

Yes. this will require a seperate API and handling mechanism. It should also be dont manually only.